### PR TITLE
Add eslint-plugin-react-hooks

### PR DESCRIPTION
**Pull request by acostalima**

### CODE REVIEW

**Comment by satazor** [comment]
The PR is good but I don’t think we need another addon for it, and simply enable it as part of the react addon. What do you think?

**Comment by acostalima** [comment]
> The PR is good but I don’t think we need another addon for it, and simply enable it as part of the react addon. What do you think?

Yeah, I guess. The reason I didn't do that right now is because there may be some projects around that still do not make use of Hooks. In such cases, activating the plugin would be for nothing. The "downside" is that developers will have to explicitly declare the add-on in their project's config.

Should we still enable the plugin in the React add-on?

**Comment by acostalima** [comment]
The pipeline is breaking in Node v6 environment. The plugin only [supports v7 onwards](https://github.com/facebook/react/blob/7a7e792a6fd7668b37e475bf39e58fdb24e4ad81/packages/eslint-plugin-react-hooks/package.json#L28). Should we drop support for v6?

**Comment by satazor** [comment]
Enabling a new plugin even if it’s not doing nothing is not that bad, it will decrease performance just a little bit.

I think this all boils down to expectations: as a developer, I expect to have hooks linting by enabling the react addon, especially because hooks are part of the official react features now.

Can you change this PR to do that? It will actually be simpler.

**Comment by satazor** [comment]
Yep we can drop it, but add it as a breaking change.

**Comment by satazor** [comment]
Also, as part of a major bump, it would be nice to update eslint to v6, but I don’t know the amount of work it will give us.

**Comment by acostalima** [comment]
> Enabling a new plugin even if it’s not doing nothing is not that bad, it will decrease performance just a little bit.
> 
> I think this all boils down to expectations: as a developer, I expect to have hooks linting by enabling the react addon, especially because hooks are part of the official react features now.
> 
> Can you change this PR to do that? It will actually be simpler.

I thought as much on the performance side of things, but I wasn't entirely sure. Anyway, I think you're right about developer's expectations.

I had something else in mind that I'd like to discuss. Since Hooks are really moving forward as part of React's ecosystem, should we consider enabling a rule in `eslint-plugin-react` to give a warning (or even an error) as an indication to favor function components instead of class components?

> Yep we can drop it, but add it as a breaking change.

Will do.

> Also, as part of a major bump, it would be nice to update eslint to v6, but I don’t know the amount of work it will give us.

That will cost us two breaking changes, but maybe we should address that in another pull request? 🤔

**Comment by satazor** [comment]
Yes lets address the eslint bump in another PR, but perhaps launch both commits in a single release.

In regards to the warning to prefer functional: we already had that enable a year ago or so, and we ended up disabling it for now. Perhaps this is something we can discuss as a team in an issue.

**Comment by acostalima** [comment]
> Yes lets address the eslint bump in another PR, but perhaps launch both commits in a single release.
> 
> In regards to the warning to prefer functional: we already had that enable a year ago or so, and we ended up disabling it for now. Perhaps this is something we can discuss as a team in an issue.

Alright! I don't think I have the needed permissions to change Travis' pipeline config. Can you do it?

**Comment by acostalima** [comment]
Nevermind my previous comment. I have just updated `.travis.yml`.

File: README.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ Then enhance it with one or more **addons**:
 - `es-modules`: If you are going to use ES6 import & export instead of CommonJS or AMD
 - `babel-parser`: Use [babel-eslint](https://github.com/babel/babel-eslint) parser so that you may use language features that are not yet implemented in eslint (e.g.: dynamic imports)
 - `react` - If you are going to use [React](https://reactjs.org/) (requires `es6` base configuration or higher)
```
Comment (satazor) [inline README.md]: This is a left over, right?

**Review by satazor — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (acostalima) [reply README.md]: You meant react-hooks? Yep, sorry!

**Review by satazor — APPROVED** [review]

**Pull request merged**